admin/app/helpers/solidus_admin/permission_sets_helper.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+module SolidusAdmin
+  module PermissionSetsHelper
```

Comment:
Although it is well crafted, I would rather not want to maintain this list.

Could we instead introduce a static class method on permission set classes that indicate it's permission type (for the lack of a better name)?

Something like

```rb
module Spree
  module PermissionSets
    class ConfigurationDisplay < PermissionSets::Base
      def self.permission_type
        :read
      end

      ...
    end
  end
end
```

That way we have this statically configured by permission set class and can get rid of the guessing game here.

Reply:
In https://github.com/solidusio/solidus/blob/e8c9675c4f15792b2a039a340232834bb9e24037/core/lib/spree/permission_sets/base.rb#L12

we could introduce that method and warn if a custom permission class has not set the name. Potentially we could even add a name match fallback to it 🤔 ?

Reply:
This might be a better approach than what we have provided in our updates to the PR so I am going to leave this comment open and unresolved until you've had a chance to take a look at our changes. (@benjaminwil helped out on some of the rewrites 💕 ) Definitely still game to rewrite and adapt to this approach if need be, but we were on a roll and got things working as expected in the end.

Reply:
Waiting for @tvdeyen opinion here before approving.

Reply:
I really like that we have control over the category and the privilege (great name 👍🏻) per persisted permission set. 

What I consider slightly problematic is that we still use class name string matching in a DB seed file to interpret what a permission set's category and privilege could be. 

I strongly believe that this should live in code in order to reduce interpretation error here. The cognitive load will be much lower if right where the permissions are set also it's category and privilege are set.

### Example

```rb
module Spree
  module PermissionSets
    class EditTaxCategories < PermissionSets::Base
      permission_category :settings
      permission_privilege :edit

      def activate!
        can [:show, :edit], Spree::TaxCategory
      end
    end
  end
end
```

This approach would have the additional advantage that we could validate privileges and categories to those we support in the components.

This is out of scope of this PR, I think. But I would like this to be a prerequisite for this PR to be merged.

Reply:
No worries, the scope of this PR is already bit bigger than I originally planned but I agree this is much better long term. I'll add these to the subclasses and fix the seeds file ✅

Reply:
Addressed everything you mentioned, thanks for the thorough reviews! Hoping I can request one last round :) @tvdeyen

Reply:
Thanks for having the patience.